### PR TITLE
bpo-32192: A basic lazy importer example

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Co-author : Nick Coghlan (ncoghlan) 

https://bugs.python.org/issue32192

### CODE REVIEW

File: Doc/library/importlib.rst [hunk 1]
```diff
@@ -1719,6 +1719,27 @@ To import a Python source file directly, use the following recipe
...
 
+Implementing lazy imports
+'''''''''''''''''''''''''
+
+The example below shows how to implement lazy imports::
+
+  import sys
+  import importlib.util
+  def lazy_import(name):
+    spec = importlib.util.find_spec(name)
```
Comment (matrixise) [inline Doc/library/importlib.rst]: Hi @nanjekyejoannah 

I like the example but could you use an indent of 4 spaces to be conform with the other examples?

Thank you,

**Review by matrixise — APPROVED** [review]
Thank you for your proposal

**Review by matrixise — CHANGES_REQUESTED** [review]
sorry, I just clicked on the wrong button, but could you use an indent of 4 spaces. Thank you

**Comment by nanjekyejoannah** [comment]
Done thanks for the review @matrixise

File: Doc/library/importlib.rst [hunk 2]
```diff
@@ -1719,6 +1719,27 @@ To import a Python source file directly, use the following recipe
...
+  def lazy_import(name):
+      spec = importlib.util.find_spec(name)
+      loader = importlib.util.LazyLoader(spec.loader)
+      spec.loader = loader
+      module = importlib.util.module_from_spec(spec)
+      sys.modules[name] = module
+      loader.exec_module(module)
+      return module
+
+  >>> lazy_typing = lazy_import("typing")
```
Comment (merwok) [inline Doc/library/importlib.rst]: Suggest adding a comment like: `lazy_typing is a real module object, but it is not loaded in memory yet`

File: Doc/library/importlib.rst [hunk 3]
```diff
@@ -1719,6 +1719,27 @@ To import a Python source file directly, use the following recipe
   spec.loader.exec_module(module)
 
 
+Implementing lazy imports
+'''''''''''''''''''''''''
+
+The example below shows how to implement lazy imports::
+
+  import sys
+  import importlib.util
```
Comment (merwok) [inline Doc/library/importlib.rst]:
```suggestion
  import importlib.util

```

File: Doc/library/importlib.rst [hunk 4]
```diff
@@ -1719,6 +1719,27 @@ To import a Python source file directly, use the following recipe
   spec.loader.exec_module(module)
 
 
+Implementing lazy imports
+'''''''''''''''''''''''''
+
+The example below shows how to implement lazy imports::
```
Comment (merwok) [inline Doc/library/importlib.rst:1725]: This is one code block with two styles (console and Python), does it render correctly?

File: Doc/library/importlib.rst (hunk 3 above)
Comment (nanjekyejoannah) [reply Doc/library/importlib.rst]: Just curious, what are you achieving with this edit?

Comment (nanjekyejoannah) [reply Doc/library/importlib.rst]: oh! I see, you added a new line. Okay makes sense.

Comment (nanjekyejoannah) [reply Doc/library/importlib.rst]: Not applicable in console mode I guess that is if you meant a new line.

File: Doc/library/importlib.rst (hunk 4 above)
Comment (nanjekyejoannah) [reply Doc/library/importlib.rst:1725]: Went with console.

**Pull request merged**